pkg/storage/filesystem/btrfs_ci_test.go

```diff
@@ -301,3 +294,56 @@ func TestBtrfsListCI(t *testing.T) {
 	ok := assert.Len(t, names, 0)
 	assert.True(t, ok, "not all pools were listed")
 }
+
+func TestCLeanUpQgroupsCI(t *testing.T) {
+	if SkipCITests {
+		t.Skip("test requires ability to create loop devices")
+	}
+
+	devices, err := SetupDevices(1)
+	require.NoError(t, err, "failed to initialize devices")
+
+	defer devices.Destroy()
+	loops := devices.Loops()
+	fs := NewBtrfs(&TestDeviceManager{loops})
+
+	names := make(map[string]struct{})
+	for idx := range loops {
+		loop := &loops[idx]
+		name := fmt.Sprintf("test-list-%d", idx)
+		names[name] = struct{}{}
+		_, err := fs.Create(context.Background(), name, pkg.Single, loop)
+		require.NoError(t, err)
+	}
+	pools, err := fs.List(context.Background(), func(p Pool) bool {
+		return strings.HasPrefix(p.Name(), "test-")
+	})
+	pool := pools[0]
+
+	_, err = pool.Mount()
+	require.NoError(t, err)
+	defer pool.UnMount()
+
+	volume, err := pool.AddVolume("vol1")
```

Comment:
I think the test here should add 2 (or more) volumes and then delete 1 and then make sure only qgroup for that volume is deleted

Reply:
yep you are right